### PR TITLE
Move UI code called in enginx systems test out of mantidqtinterfaces

### LINKED ISSUE #33094: Move code called in ENGINX systems test from focus and calibration models to directory outside mantidqtinterfaces
Building an package/.exe for mantid with flags `MANTID_QT_LIB=OFF` and `ENABLE_WORKBENCH=OFF` means that code inside mantidqtinterfaces will not be present. In PR #32823 the ENGINX systems test was refactored to call the Engineering Diffraction GUI code (in mantidqtinterfaces) so it would not be possible to run the systems test with the above flags set to `OFF` (get an import error).

To fix this it is necessary to move the code called by the systems test to a directory outside mantidqtinterfaces e.g.
`C:\mantid\scripts\Engineering\common\`

The plan is to move the `output_path` and possibly the `settings_helper` modules to the above directory and put all the calibration and focus model functions inside `EnggUtils.py` (the model class will have the same methods but will just call the functions in EnggUtils)

### PR DESCRIPTION
**Pull request by RichardWaiteSTFC**
**Description of work.**

See original issue (#33094) for background.
This PR has moved code around such that systems test and UI call same code that is not in mantidqtinterfaces.

**To test:**
This has touched a lot of files so best to test all tabs of the UI as well as code review
(1) Run through manual testing instructions 
https://developer.mantidproject.org/Testing/EngineeringDiffraction/EngineeringDiffractionTestGuide.html
(2) Run the systems test on linux
`systemtest.bat -C debug -R EnginXScriptTest -j 2`

Fixes #33094 

*This does not require release notes* because **it is an internal change that shouldn't be visible to users**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Engineering/EnggUtils.py [hunk 1]
```diff
@@ -116,20 +130,448 @@ def default_ceria_expected_peaks(final=False):
...
+            mantid.SaveNexus(InputWorkspace=sample_ws_foc, Filename=nxs_path, WorkspaceIndexList=[ispec])
+            nxs_paths.append(nxs_path)
+    return nxs_paths  # from last focus_dir only
+
+
+def _generate_output_file_name(inst, sample_run_no, van_run_no, suffix, xunit, ext=""):
+    return "_".join([inst, sample_run_no, van_run_no, suffix, xunit]) + ext
+
+
+# DEPRECATED FUNCTIONS BELOW
```
Comment (DannyHindson) [inline scripts/Engineering/EnggUtils.py:568]: Good job on separating these out. Will make it easier to delete stuff when remove the deprecated algorithms

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/presenter.py [hunk 2]
```diff
@@ -18,7 +17,7 @@
 from .settings.settings_model import SettingsModel
 from .settings.settings_view import SettingsView
 from .settings.settings_presenter import SettingsPresenter
-from .settings.settings_helper import get_setting, set_setting
+from mantidqtinterfaces.Engineering.gui.engineering_diffraction.settings.settings_helper import get_setting, set_setting
```
Comment (DannyHindson) [inline qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/presenter.py:20]: This is nit picking really - but does this line need changing? The other imports seem to work with a relative path

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/settings/settings_presenter.py [hunk 3]
```diff
@@ -129,9 +129,12 @@ def _validate_settings(self):
         self.check_and_populate_with_default("default_peak")
         if not self.settings["default_peak"] in ALL_PEAKS:
             self.settings["default_peak"] = DEFAULT_SETTINGS["default_peak"]
+        self.check_and_populate_with_default("full_calibration")
+        if not path.isfile(self.settings["full_calibration"]):
+            self.settings["full_calibration"] = DEFAULT_SETTINGS["full_calibration"]
```
Comment (DannyHindson) [inline qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/settings/settings_presenter.py:134]: Nice addition - I guess you found it was failing silently\unpleasantly if given an invalid path?

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/calibration/model.py [hunk 4]
```diff
@@ -4,277 +4,27 @@
...
-    CreateEmptyTableWorkspace, NormaliseByCurrent, ConvertUnits, Load, SaveNexus, ApplyDiffCal
+from mantid.simpleapi import Load, AnalysisDataService as ADS, logger
 import Engineering.EnggUtils as EnggUtils
-from mantidqtinterfaces.Engineering.gui.engineering_diffraction.settings.settings_helper import get_setting
-from mantidqtinterfaces.Engineering.gui.engineering_diffraction.settings.settings_presenter import CALIB_FOLDER
 from mantidqtinterfaces.Engineering.gui.engineering_diffraction.tabs.common import output_settings
-from Engineering.common import path_handling
-
-DIFF_CONSTS_TABLE_NAME = "diffractometer_consts_table"
+from mantidqtinterfaces.Engineering.gui.engineering_diffraction.settings.settings_helper import get_setting
```
Comment (DannyHindson) [inline qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/calibration/model.py:10]: This comment doesn't specifically relate to this line but just needed to attach it somewhere - might be worth adding a comment in this unit somewhere to explain the model shares code with the EnginX script unit that's used for auto reduction. The reader might work that out by following the imports but would make it quicker to understand

**Review by DannyHindson — COMMENTED** [review]
I've been through the code changes and just added a couple of v minor things. Hardly worth bother with on their own but nice to have if you've got time.

Have you had chance to test if a build now works with the CMake flags to generate the UI turned off? They are MANTID_QT_LIB and ENABLE_WORKBENCH I think which can both be set to OFF

**Comment by DannyHindson** [comment]
I did the test on the CMake flags and built an installer myself just now and it works fine. I ran this from the python prompt from the python.exe in the install:
```
from Engineering.EnginX import EnginX
from Engineering.EnggUtils import GROUP

testE = EnginX(vanadium_run="ENGINX236516", focus_runs=["ENGINX299080"], full_inst_calib_path="C:\MantidUnstableInstall\scripts\Engineering\calib\ENGINX_full_instrument_calibration_193749", ceria_run="ENGINX193749", group=GROUP.BOTH, save_dir="C:\MantidUnstableInstall\scripts\Engineering\save")
```

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/presenter.py (hunk 2 above)
Comment (RichardWaiteSTFC) [reply qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/presenter.py:20]: Yep - I think that import line was added automatically by pycharm - I'll make it relative

**Review by DannyHindson — APPROVED** [review]
Good job! Nice to have this all tidied up now

**Pull request merged**